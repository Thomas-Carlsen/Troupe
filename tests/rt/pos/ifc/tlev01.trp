let val x = 42 raisedTo {alice}
    val y = 10 raisedTo {bob}
    val c = 100 raisedTo {charlie}
    val d = 200 raisedTo {dorothy}
    (* val z = let pini authority val x = if x + y > 100 then 10 else 20 in x end  *)

    val a = 
            let pini attenuate (authority, {charlie}) val a = 
                  if c > 10 then 
                            x + y
                         else 
                            x - y
            in a end 

    val _ = printWithLabels a   
    val _ = x + y       
    val _ = adv ()

    

    val p = if d > 100 then a + 1 else 0

    val _ = printWithLabels p

    (* val w = x + y  *)
    (* val y = declassify (z, authority, {}) *)
    (* val _ = printWithLabels y *)
    (* val _ = printWithLabels z *)
    (* val _ = printWithLabels w *)
in ()
end
