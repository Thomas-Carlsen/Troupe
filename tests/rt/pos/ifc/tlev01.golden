2019-03-07T14:26:48.447Z [RTM] [32minfo[39m: Skipping network creation. Observe that all external IO operations will yield a runtime error.
52@{alice,bob,charlie}%{charlie}
53@{alice,bob,charlie,dorothy}%{charlie,dorothy}
>>> Main thread finished with value: ()@{dorothy,charlie}%{dorothy,charlie}
